libkmod/libkmod-index.c

```diff
@@ -628,18 +628,14 @@ struct index_mm_value {
 	const char *value;
 };
 
-struct index_mm_value_array {
-	struct index_mm_value *values;
-	size_t len;
-};
-
 struct index_mm_node {
 	struct index_mm *idx;
 	const char *prefix; /* mmap'ed value */
-	struct index_mm_value_array values;
 	unsigned char first;
 	unsigned char last;
-	uint32_t children[];
+	const void *children; /* mmap'ed value */
+	size_t value_count;
+	const void *values; /* mmap'ed value */
```

Comment:
You cannot just increment this pointer though. `values[0]` might work but don't let anyone believe that `values[1]` would even end up at the right position, because these values are dynamic in size. Also, just as above, the numbers are big endian here. Make it clear again that these pointers must be treated specially.

Reply:
Yes, there the values (aka struct index_mm_value) is varying sized, so using `struct index_mm_value *` as type will be confusing.